### PR TITLE
feat: add on this day from wiki

### PR DESCRIPTION
**Pull request by ellenc345**
you can see it in the last card in profile page :)

### CODE REVIEW

File: app/routers/profile.py [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
 from app import config
 from app.database.database import get_db
 from app.database.models import User
 from app.dependencies import MEDIA_PATH, templates
-
+from app.internal.on_this_day_events import get_on_this_day_events
+from fastapi import APIRouter, Depends, File, Request, UploadFile
+from PIL import Image
+from starlette.responses import RedirectResponse
+from starlette.status import HTTP_302_FOUND
```
Comment (Gonzom) [inline app/routers/profile.py:11]: Your import sorting setting isn't configured correctly.

Comment (ellenc345) [reply app/routers/profile.py:11]: can you please helpme to find the right config? or i can just revrt the sorting @Gonzom

Comment (Gonzom) [reply app/routers/profile.py:11]: I can help with PyCharm, as that is what I'm using.
1. Go to File-> Setting (ctrl+alt+s)
2. Editor
3. Code Style
4. Python
5. Go to the Imports tab
6. Make sure the following are marked: 
6.1. Sort import statements
6.2 Sort import statements in "from" imports
6.3 Sort case-insensitively (not sure about this one)

Comment (ellenc345) [reply app/routers/profile.py:11]: Im using vscode, i cant find those settings,
i see only python.sortImports.args that i can pass but its not seems to be the setting you mentioned.
@Gonzom

File: app/internal/on_this_day_events.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import json
+from datetime import datetime
+
+import requests
+
+
+def get_on_this_day_events():
+    now = datetime.now()
+    day = now.day
```
Comment (ivarshav) [inline app/internal/on_this_day_events.py:9]: Could save a line: :)
`day, month = now.day, now.month`

File: app/internal/on_this_day_events.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    month = now.month
+
+    res = requests.get(
```
Comment (yammesicka) [inline app/internal/on_this_day_events.py:12]: Prefer prefetching this one time only.

Populate the database with the relevant info using some script.
Running `requests.get` on our server is a blocking call, and will probably not perform well on multiple-users environment.

File: app/internal/on_this_day_events.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import json
+from datetime import datetime
+
+import requests
+
+
+def get_on_this_day_events():
```
Comment (yammesicka) [inline app/internal/on_this_day_events.py:7]: Add type annotations

**Review by yammesicka — CHANGES_REQUESTED** [review]
Good job! Please see my insights about the possible performance issues.

**Comment by ellenc345** [comment]
@yammesicka You totaly right, i changed the base branch to develop and added db cashing. pls review again :)

File: app/internal/on_this_day_events.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    res_wiki = text.get('wikipedia')
+    session.add(WikipediaEvents(events=res_events,
+                                date_=res_date, wikipedia=res_wiki))
+    session.commit()
+    return text
+
+
+def get_on_this_day_events(session: Session) -> Dict:
+    try:
+        data = session.query(WikipediaEvents).\
```
Comment (yammesicka) [inline app/internal/on_this_day_events.py:31]: Prefer splitting using parenthesis, not `\`

File: app/internal/on_this_day_events.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+            filter(func.date(WikipediaEvents.date_inserted) == date.today()).\
+            one()
+
+    except NoResultFound:
+        data = insert_on_this_day_data(session)
+    except (SQLAlchemyError, AttributeError) as e:
+        print('err', e)
```
Comment (yammesicka) [inline app/internal/on_this_day_events.py:38]: Use our logging system

**Review by yammesicka — COMMENTED** [review]
Awesome job on this one! Please add tests to insert_on_this_day_data and get_on_this_day_events

File: tests/test_on_this_day_events.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    assert type(data.get('wikipedia')) == str
+
+
+def test_get_on_this_day_events_exists(session):
+    fake_object = WikipediaEvents(
+        events=['fake'], wikipedia="www.fake.com", date_="not a date string")
+    session.add(fake_object)
+    session.commit()
+    fake_data = get_on_this_day_events(session)
+    print(fake_data.events)
```
Comment (yammesicka) [inline tests/test_on_this_day_events.py:27]: Remove the `print` here

File: tests/test_on_this_day_events.py [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    is_exists_data = session.query(WikipediaEvents).all()
+    assert not is_exists_data
+    insert_on_this_day_data(session)
+    is_exists_data = session.query(WikipediaEvents).all()
+    assert is_exists_data is not None
+
+
+def test_get_on_this_day_events(session):
+    data = get_on_this_day_events(session)
+    assert type(data) == dict
```
Comment (yammesicka) [inline tests/test_on_this_day_events.py:16]: Prefer `isinstance`

File: app/templates/profile.html [hunk 9]
```diff
@@ -273,9 +273,7 @@ <h6 class="card-title text-center mb-1">{{ user.full_name }}</h6>
                 <!-- Card -->
                 <div class="card card-profile mb-3" style="height: 18em;">
                     <div class="card-body">
-                        <p class="card-title">
-                            Your Card
-                        </p>
+                        {% include "./on_this_day.html" %}
```
Comment (yammesicka) [inline app/templates/profile.html:276]: Is the `./` needed?

File: app/routers/profile.py [hunk 10]
```diff
@@ -1,15 +1,16 @@
-import io
-
-from fastapi import APIRouter, Depends, File, Request, UploadFile
-from starlette.responses import RedirectResponse
 from starlette.status import HTTP_302_FOUND
```
Comment (yammesicka) [inline app/routers/profile.py:1]: Sort alphabetically, move io to be in the first paragraph and app.internal to be in the third one

File: app/internal/on_this_day_events.py [hunk 11]
```diff
@@ -0,0 +1,42 @@
+import json
+from datetime import date, datetime
+from typing import Dict
+
+import requests
+from app.database.models import WikipediaEvents
```
Comment (yammesicka) [inline app/internal/on_this_day_events.py:6]: This line should be in a separate 3rd paragraph

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: app/templates/profile.html (hunk 9 above)
Comment (ellenc345) [reply app/templates/profile.html:276]: surprisingly, not needed. thank you

**Comment by ellenc345** [comment]
passed the tests!

File: app/internal/on_this_day_events.py [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+from loguru import logger
+from sqlalchemy import func
+from sqlalchemy.exc import SQLAlchemyError
+from sqlalchemy.orm import Session
+from sqlalchemy.orm.exc import NoResultFound
+
+from app.database.models import WikipediaEvents
+
+
+def insert_on_this_day_data(session: Session) -> Dict:
```
Comment (yammesicka) [inline app/internal/on_this_day_events.py:15]: Use `Depends(get_db)` instead. Also, in the type annotations, add types for the `Dict`'s key and value.

File: app/internal/on_this_day_events.py [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+    res_events = text.get('events')
+    res_date = text.get('date')
+    res_wiki = text.get('wikipedia')
+    session.add(WikipediaEvents(events=res_events,
+                                date_=res_date, wikipedia=res_wiki))
+    session.commit()
+    return text
+
+
+def get_on_this_day_events(session: Session) -> Dict:
```
Comment (yammesicka) [inline app/internal/on_this_day_events.py:31]: Same as `insert_on_this_day_data`

**Review by yammesicka — COMMENTED** [review]
Great job! Just another little push

**Review by yammesicka — APPROVED** [review]

**Pull request merged**